avaje-config/src/main/java/io/avaje/config/CoreConfigurationBuilder.java

```diff
@@ -1,35 +1,56 @@
 package io.avaje.config;
 
-import io.avaje.lang.NonNullApi;
+import static java.lang.System.Logger.Level.DEBUG;
+import static java.lang.System.Logger.Level.INFO;
+import static java.util.Objects.requireNonNull;
 
 import java.io.File;
 import java.io.FileReader;
 import java.io.IOException;
 import java.io.UncheckedIOException;
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Map;
 import java.util.Properties;
 import java.util.ServiceLoader;
-import java.util.stream.Collectors;
 
-import static java.lang.System.Logger.Level.DEBUG;
-import static java.lang.System.Logger.Level.INFO;
-import static java.util.Objects.requireNonNull;
+import io.avaje.config.CoreConfiguration.ForegroundEventRunner;
+import io.avaje.lang.NonNullApi;
 
 @NonNullApi
 final class CoreConfigurationBuilder implements Configuration.Builder {
 
-  private ConfigurationLog log = initialiseLog();
+  private static ConfigurationLog spiLog;
+  private static ResourceLoader spiResourceLoader;
+  private static ModificationEventRunner spiEventRunner;
+  private static final List<ConfigurationSource> SOURCES= new ArrayList<>();
+  private static final List<ConfigurationPlugin> PLUGINS= new ArrayList<>();
+
+  private ConfigurationLog log = spiLog;
   private final Parsers parsers = new Parsers();
   private final CoreEntry.CoreMap sourceMap = CoreEntry.newMap();
-  private ResourceLoader resourceLoader = initialiseResourceLoader();
-  private ModificationEventRunner eventRunner;
+  private ResourceLoader resourceLoader = spiResourceLoader;
+  private ModificationEventRunner eventRunner = spiEventRunner;
   private boolean includeResourceLoading;
   private InitialLoader initialLoader;
 
-  private static ConfigurationLog initialiseLog() {
-    return ServiceLoader.load(ConfigurationLog.class)
-        .findFirst()
-        .orElseGet(DefaultConfigurationLog::new);
+  static {
```

Comment:
I was expecting for there to be a separate class that owned all the service loading. Hmmm.